arch/cpu/native/Makefile.native

```diff
@@ -15,7 +15,10 @@ NM       ?= nm
 OBJCOPY  ?= objcopy
 STRIP    ?= strip
 ifeq ($(WERROR),1)
-CFLAGSWERROR=-Werror
+CFLAGSWERROR +=-Werror 
+CFLAGSWERROR +=-Wno-error=unused-function -Wno-unused-function
+# printf( (char*)pfmt ) - here fmt is variable, may warns on modern compiler 
```

Comment:
Please remove the commented out lines, then I can merge the PR. 
There is no need to close and resubmit, just add the change in a new commit and `git rebase -i contiki-orig/develop`. Then select "fixup" for the new commit to include it in an existing one, and then `git push -f origin`
See https://github.com/contiki-os/contiki/wiki/Code-Contributions#how-to-contribute-code